src/hdmf/utils.py

```diff
@@ -634,19 +634,19 @@ def __get_shape_helper(local_data):
             if len(local_data) and not isinstance(local_data[0], (str, bytes)):
                 shape.extend(__get_shape_helper(local_data[0]))
         return tuple(shape)
+
+    shape = np.shape(data)
+    if shape:
+        return shape
     if hasattr(data, 'maxshape'):
```

Comment:
To maintain existing behavior where maxshape is checked before shape, this check should go before `np.shape`